src/mesh/X3D_Draco_Mesh_Reader.cc

```diff
@@ -130,6 +132,31 @@ void X3D_Draco_Mesh_Reader::read_mesh() {
   x3d_cellface_map = map_x3d_block<int, int>("cells", dist);
   Check(dist > dist_old);
 
+  // parse x3d material flags
+  while (dist < parsed_pairs.size() && parsed_pairs[dist].first != "matid")
+    dist++;
+
+  if (parsed_pairs[dist].first != "matid") {
+    throw std::invalid_argument("no matid found in x3d file " + filename);
+  }
+
+  dist++;
+  // As the X3D manual notes, matid is a little weird.
+  x3d_matids.resize(get_numcells());
+  unsigned i = 0;
+  do {
+    if (dist > parsed_pairs.size()) {
+      throw std::invalid_argument("wrong number of matids in x3d file " +
+                                  filename);
+    }
+    x3d_matids[i++] = parsed_pairs[dist].first;
+    auto const &matids = parsed_pairs[dist].second;
+    dist++;
+    for (auto const &id : matids) {
+      x3d_matids[i++] = id;
+    }
+  } while (i < x3d_matids.size());
```

Comment:
Can we change do ... while to just a while loop?

Reply:
Yes. Actually, that might be a bit clearer. Will make the change.